src/ts/features/block-navigation-mode/index.ts

```diff
@@ -0,0 +1,392 @@
+            },
+        }),
+        nmap({
+            id: 'panelRight',
+            key: 'l',
+            label: 'Select Panel Right',
+            onPress: () => {
+                if (document.querySelector(Selectors.sidebarContent)) {
+                    state.panel = 'SIDEBAR'
+                }
+            },
+        }),
+        nmap({
+            id: 'clickSelection',
+            key: 'i',
+            label: 'Click Selection',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Mouse.leftClick(block as HTMLElement)
+                    Roam.moveCursorToStart()
+                }
+            },
+        }),
+        nmap({
+            id: 'clickSelectionAndGotoEnd',
+            key: 'a',
+            label: 'Click Selection and Go-to End of Line',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Mouse.leftClick(block as HTMLElement)
+                    Roam.moveCursorToEnd()
+                }
+            },
+        }),
+        map({
+            id: 'closeSplitPage',
+            key: 'Control+w',
+            label: 'Close Page in Side Bar',
+            updateView: false,
+            onPress: () => {
+                const block = selectedBlock()
+                if (block) {
+                    const pageContainer = block.closest(`${Selectors.sidebarContent} > div`)
+                    const closeButton = pageContainer?.querySelector(Selectors.closeButton)
+                    if (closeButton) {
+                        Mouse.leftClick(closeButton as HTMLElement)
+                    }
+                }
+            },
+        }),
+        nmap({
+            id: 'insertBlockBefore',
+            key: 'O',
+            label: 'Insert Block Before',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Roam.activateBlock(block)
+                    await Roam.moveCursorToStart()
+                    await Keyboard.pressEnter()
+                }
+            },
+        }),
+        nmap({
+            id: 'insertBlockAfter',
+            key: 'o',
+            label: 'Insert Block After',
+            onPress: async () => {
+                await insertBlockAfter()
+            },
+        }),
+        nmap({
+            id: 'toggleFold',
+            key: 'z',
+            label: 'Toggle Fold Block',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Roam.toggleFoldBlock(block)
+                }
+            },
+        }),
+        nmap({
+            id: 'enterVisualMode',
+            key: 'V',
+            label: 'Enter Visual Mode',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
```

Comment:
this is repeated often, would probably makes sense a function like `applyToSelected` that would execute supplied function only if the block is present. Potentially also abstracting away the "returnToNormalMode"

Reply:
I decided to fallback to selecting the first block, if no block is selected for some reason, which circumvents the null checks.